packages/core/lib/config-migration.js

```diff
@@ -0,0 +1,47 @@
+const OS = require("os");
+const path = require("path");
+const fse = require("fs-extra");
+const Conf = require("conf");
+const { promisify } = require("util");
+const copy = require("./copy");
+const TruffleConfig = require("@truffle/config");
+
+module.exports = {
+  oldTruffleDataDirectory: path.join(OS.homedir(), ".config", "truffle"),
+
+  needsMigrated: function () {
+    const oldConfig = path.join(this.oldTruffleDataDirectory, "config.json");
+    const conf = new Conf({ projectName: "truffle" });
+    return fse.existsSync(oldConfig) && oldConfig !== conf.path;
```

Comment:
if adding the debug module, would be nice to log if it needs migration

Reply:
Should there be another component to this logic? Stuff gets copied the first time, and there's not delete (or rename) of the original so the second time a command runs,
`fse.existsSync(oldConfig) && oldConfig !== conf.path` will be true again? Am I missing something?